### PR TITLE
C387477 Browsing call number types when "Call numbers (all)" browse option selected (based on "Item") (spitfire) (TaaS)

### PR DESCRIPTION
**Pull request by artsiomBandarenka**
[C387477](https://foliotest.testrail.io/index.php?/cases/view/387477)

[report](https://jenkins-aws.indexdata.com/job/folioRancher/job/folioTestingTools/job/runCypressTests/4894/allure/)
image

### CODE REVIEW

File: cypress/e2e/inventory/search/browse-call-numbers-all.cy.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+import InventoryInstances from '../../../support/fragments/inventory/inventoryInstances';
+import InventorySearchAndFilter from '../../../support/fragments/inventory/inventorySearchAndFilter';
+import Location from '../../../support/fragments/settings/tenant/locations/newLocation';
+import ServicePoints from '../../../support/fragments/settings/tenant/servicePoints/servicePoints';
+import { BROWSE_CALL_NUMBER_OPTIONS } from '../../../support/constants';
+import InventoryInstance from '../../../support/fragments/inventory/inventoryInstance';
+import ItemRecordNew from '../../../support/fragments/inventory/item/itemRecordNew';
+import BrowseCallNumber from '../../../support/fragments/inventory/search/browseCallNumber';
+
+describe('Call Number Browse', () => {
```
Comment (manowarqqq) [inline cypress/e2e/inventory/search/browse-call-numbers-all.cy.js:14]: Please, use two describe. One for 'Inventory', second one for Call Number Browse'

File: cypress/e2e/inventory/search/browse-call-numbers-all.cy.js [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        cy.login(userProperties.username, userProperties.password, {
+          path: TopMenu.inventoryPath,
+          waiter: InventoryInstances.waitContentLoading,
+        });
+      })
+      .then(() => {
+        InventoryInstance.searchByTitle(instance.title);
+        InventoryInstances.selectInstance();
+        callNumbers.forEach((el, index) => {
+          InventoryInstance.addItem();
```
Comment (manowarqqq) [inline cypress/e2e/inventory/search/browse-call-numbers-all.cy.js:89]: Probably could be done with ItemRecordNew.createViaApi()

**Review by zentestuken — APPROVED** [review]

File: cypress/support/fragments/inventory/inventorySearchAndFilter.js [hunk 3]
```diff
@@ -685,4 +686,15 @@ export default {
...
   },
+
+  browseOptionsDropdownIncludesOptions(options) {
+    const browseOptionsDropdown = Select('Search field index');
+    cy.do(browseButton.click());
+    // cypress can't draw selected option without wait
+    // eslint-disable-next-line cypress/no-unnecessary-waiting
+    cy.wait(ONE_SECOND);
+    options.forEach((name) => {
+      cy.expect(browseOptionsDropdown.find(Option(name)).exists());
```
Comment (manowarqqq) [inline cypress/support/fragments/inventory/inventorySearchAndFilter.js:697]: Should work without timeout with 
      cy.expect(browseOptionsDropdown.has({ content: including(name) }));

File: cypress/e2e/inventory/search/browse-call-numbers-all.cy.js (hunk 1 above)
Comment (artsiomBandarenka) [reply cypress/e2e/inventory/search/browse-call-numbers-all.cy.js]: done

File: cypress/e2e/inventory/search/browse-call-numbers-all.cy.js (hunk 2 above)
Comment (artsiomBandarenka) [reply cypress/e2e/inventory/search/browse-call-numbers-all.cy.js]: yes, done

File: cypress/support/fragments/inventory/inventorySearchAndFilter.js (hunk 3 above)
Comment (artsiomBandarenka) [reply cypress/support/fragments/inventory/inventorySearchAndFilter.js]: done

**Comment by artsiomBandarenka** [comment]
> Please, replace allure report with valid link and add screenshot from this report because this link will be expired soon Please use rebase instead of merge to get 1 commit for merging

done

**Review by manowarqqq — APPROVED** [review]

**Pull request merged**